bitcoin/src/psbt/mod.rs

```diff
@@ -385,6 +397,79 @@ impl Psbt {
         Ok(used)
     }
 
+    /// Attempts to create all signatures required by this PSBT's `tap_key_origins` field, adding
+    /// them to `tap_key_sig` or `tap_script_sigs`.
+    ///
+    /// # Returns
+    ///
+    /// - Ok: A list of the public keys used in signing.
+    /// - Err: Error encountered trying to calculate the sighash AND we had the signing key.
+    fn bip32_sign_schnorr<C, K, T>(
+        &mut self,
+        k: &K,
+        input_index: usize,
+        cache: &mut SighashCache<T>,
+        secp: &Secp256k1<C>,
+    ) -> Result<Vec<PublicKey>, SignError>
+        where
+            C: Signing,
+            T: Borrow<Transaction>,
+            K: GetKey,
+    {
+
+        let mut input = self.inputs[input_index].clone();
+
+        let mut used = vec![]; // List of pubkeys used to sign the input.
+
+        for (xpk, (leaf_hashes, key_source)) in input.tap_key_origins.iter() {
+            let (sk, x_only_pubkey) = if let Ok(Some(sk)) = k.get_key(KeyRequest::Bip32(key_source.clone()), secp) {
+                (sk, *xpk)
+            } else {
+                continue;
+            };
+
+            //key path spend
+            if let Some(internal_key) = input.tap_internal_key {
+                if internal_key == x_only_pubkey && leaf_hashes.len() == 0 &&input.tap_key_sig.is_none() {
+                    let (msg, sighash_ty) = self.sighash_taproot(input_index, cache, None)?;
+                    let secp_for_tweak = &Secp256k1::<secp256k1::All>::gen_new();
+                    let key_pair = Keypair::from_secret_key(secp_for_tweak, &sk.inner)
+                        .tap_tweak(secp_for_tweak, input.tap_merkle_root)
+                        .to_inner();
+                    let final_signature = taproot::Signature { signature: secp.sign_schnorr_no_aux_rand(&msg, &key_pair), sighash_type: sighash_ty };
+                    input.tap_key_sig = Some(final_signature);
+                }
+            }
+
+            //script path spend
+            if let Some((leaf_hashes, _)) = input.tap_key_origins.get(&x_only_pubkey) {
+                let leaf_hashes = leaf_hashes
+                    .iter()
+                    .filter(|lh| {
+                        !input
+                            .tap_script_sigs
+                            .contains_key(&(x_only_pubkey, **lh))
+                    })
+                    .cloned()
+                    .collect::<Vec<_>>();
+
+                for lh in leaf_hashes {
+                    let (msg, sighash_ty) = self.sighash_taproot(input_index, cache, Some(lh))?;
+                    let key_pair = Keypair::from_secret_key(secp, &sk.inner);
+                    let final_signature = taproot::Signature { signature: secp.sign_schnorr_no_aux_rand(&msg, &key_pair), sighash_type: sighash_ty };
```

Comment:
In this case, I propose to use the presence of the rand-std feature as the condition to determine whether to utilize aux rand. Since there is no suitable random source available in the no-std environment, the assumption is that aux rand is only available in the std environment. 
I would appreciate your feedback on this approach and any suggestions or recommendations you may have.